build/lib/release.sh

```diff
@@ -274,41 +301,33 @@ function kube::release::create_docker_images_for_server() {
 
       local binary_name="$1"
       local base_image="$2"
+      local docker_build_path="${binary_dir}/${binary_name}.dockerbuild"
+      local docker_file_path="${docker_build_path}/Dockerfile"
+      local binary_file_path="${binary_dir}/${binary_name}"
+      local docker_image_tag="${docker_registry}"
+      if [[ ${arch} == "amd64" ]]; then
+        # If we are building a amd64 docker image, preserve the original
+        # image name
+        docker_image_tag+="/${binary_name}:${docker_tag}"
+      else
+        # If we are building a docker image for another architecture,
+        # append the arch in the image tag
+        docker_image_tag+="/${binary_name}-${arch}:${docker_tag}"
+      fi
 
-      kube::log::status "Starting Docker build for image: ${binary_name}"
 
+      kube::log::status "Starting docker build for image: ${binary_name}-${arch}"
       (
-        # Docker tags cannot contain '+'
-        local docker_tag="${KUBE_GIT_VERSION/+/_}"
-        if [[ -z "${docker_tag}" ]]; then
-          kube::log::error "git version information missing; cannot create Docker tag"
-          return 1
-        fi
-        local docker_build_path="${binary_dir}/${binary_name}.dockerbuild"
-        local docker_file_path="${docker_build_path}/Dockerfile"
-        local binary_file_path="${binary_dir}/${binary_name}"
-        local docker_image_tag="${KUBE_DOCKER_REGISTRY:-gcr.io/google_containers}"
-
         rm -rf ${docker_build_path}
         mkdir -p ${docker_build_path}
         ln ${binary_dir}/${binary_name} ${docker_build_path}/${binary_name}
         printf " FROM ${base_image} \n ADD ${binary_name} /usr/local/bin/${binary_name}\n" > ${docker_file_path}
 
-        if [[ ${arch} == "amd64" ]]; then
-          # If we are building a amd64 docker image, preserve the original
-          # image name
-          docker_image_tag+="/${binary_name}:${docker_tag}"
-        else
-          # If we are building a docker image for another architecture,
-          # append the arch in the image tag
-          docker_image_tag+="/${binary_name}-${arch}:${docker_tag}"
-        fi
-
         "${DOCKER[@]}" build --pull -q -t "${docker_image_tag}" ${docker_build_path} >/dev/null
-        "${DOCKER[@]}" save ${docker_image_tag} > ${binary_dir}/${binary_name}.tar
+        "${DOCKER[@]}" save "${docker_image_tag}" > "${binary_dir}/${binary_name}.tar"
         echo "${docker_tag}" > ${binary_dir}/${binary_name}.docker_tag
-
         rm -rf ${docker_build_path}
+        ln "${binary_dir}/${binary_name}.tar" "${images_dir}/"
```

Comment:
yeah, the docker tarfiles are pretty large. for amd64 alone, ~576M for everything except hyperkube, and another 582M for hyperkube. across 5 architectures that seemed like a lot.